cinema/views.py

```diff
@@ -2,0 +1,65 @@
+from rest_framework import viewsets
+
+from cinema.models import Genre, Actor, Movie, MovieSession, CinemaHall
+from cinema.serializers import (
+    GenreSerializer,
+    ActorSerializer,
+    MovieSerializer, MovieSessionSerializer, CinemaHallSerializer,
+    MovieListSerializer, MovieRetrieveSerializer,
+    MovieSessionRetrieveSerializer,
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+
+        return MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            return queryset.prefetch_related("genres", "actors")
+        return queryset
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+    serializer_class = MovieSessionSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieSessionSerializer
+        elif self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+        return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            return queryset.select_related(
+                "movie", "cinema_hall"
+            ).prefetch_related(
+                "movie__genres", "movie__actors"
+            )
+
+
+class CinemaHallsViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
```

Comment:
Consider removing the `.all()` method before using `.prefetch_related()` or `.select_related()` to ensure efficient database queries. This is a best practice according to the task checklist.